readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/hive/HiveClient.java

```diff
@@ -16,10 +19,15 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(HiveClient.class);
 
     private final HiveQueryExecutor executor;
-    private final Set<Pair<String, String>> createdPartitions = new HashSet<>();
+    private final Set<Pair<String, String>> createdPartitions;
     private final Set<String> createdTables = new HashSet<>();
 
     public HiveClient(HiveQueryExecutor executor, String location) throws SQLException {
+        this.createdPartitions = Collections.newSetFromMap(new LinkedHashMap<Pair<String, String>, Boolean>(){
+            protected boolean removeEldestEntry(Entry<Pair<String, String>, Boolean> eldest) {
+                return size() > 100;
```

Comment:
Cool, I did not know about this method !

I feel like 100 is enough but could you just explain your rational for this number ? Because we could easily go up to thousands to avoid any surprise as the goal is mostly to prevent out-of-memory issues after days of run

Reply:
However I agree that is it best effort to prevent to many requests...

Reply:
AFAIR, there are 8 event types and usually a 2-day window. So theoretically even 16 should be enough. It's a protective cache anyway, so it doesn't hurt if a query slips through. We mostly want to avoid the bulk, so even 8 could be sufficient.

Reply:
So there's no strong rationale for 100. I could have picked a power of 2 as well, wouldn't have more sense :smile:

Reply:
lol